src/commands/version.ts

```diff
@@ -22,55 +12,50 @@ export default class Version extends Command {
     }),
   }
 
-  async run(): Promise<VersionDetail> {
+  async run(): Promise<Partial<Interfaces.VersionDetails>> {
     const {flags} = await this.parse(Version)
 
-    const [cliVersion, architecture, nodeVersion] = this.config.userAgent.split(' ')
-
-    const versionDetail:VersionDetail = {cliVersion, architecture, nodeVersion}
     let output = `${this.config.userAgent}`
-
     if (flags.verbose) {
-      const pluginVersions = []
-      for (const plugin of this.config.plugins.sort((a, b) => a.name > b.name ? 1 : -1)) {
-        if (this.config.name !== plugin.name) {
-          pluginVersions.push(`${this.getFriendlyName(plugin.name)} ${plugin.version} (${plugin.type}) ${plugin.type === 'link' ? plugin.root : ''}`.trim())
-        }
-      }
-
-      const osVersion = `${osType()} ${osRelease()}`
-
-      versionDetail.pluginVersions = pluginVersions
-      versionDetail.osVersion = osVersion
-      versionDetail.shell = this.config.shell
-      versionDetail.rootPath = this.config.root
-
       output = ` CLI Version:
-\t${cliVersion}
+\t${this.config.versionDetails.cliVersion}
 
  Architecture:
-\t${architecture}
+\t${this.config.versionDetails.architecture}
 
  Node Version:
-\t${nodeVersion}
+\t${this.config.versionDetails.nodeVersion}
 
  Plugin Version:
-\t${pluginVersions.join(`${EOL}\t`)}
+\t${flags.verbose ? this.formatPlugins(this.config.versionDetails.pluginVersions ?? {}).join(EOL + '\t') : ''}
 
  OS and Version:
-\t${osVersion}
+\t${this.config.versionDetails.osVersion}
 
  Shell:
-\t${versionDetail.shell}
+\t${this.config.versionDetails.shell}
 
  Root Path:
-\t${versionDetail.rootPath}
+\t${this.config.versionDetails.rootPath}
 `
     }
 
     this.log(output)
 
-    return versionDetail
+    return flags.verbose ?
```

Comment:
the shape of the verbose json has changed. So probably a breaking change. I'll bump pjson version to next major.